sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/TryCastSuite.scala

```diff
@@ -17,30 +17,50 @@
 
 package org.apache.spark.sql.catalyst.expressions
 
+import scala.reflect.ClassTag
+
 import org.apache.spark.SparkFunSuite
 import org.apache.spark.sql.catalyst.InternalRow
 import org.apache.spark.sql.catalyst.util.DateTimeTestUtils.UTC_OPT
 import org.apache.spark.sql.types._
 import org.apache.spark.unsafe.types.UTF8String
 
 // A test suite to check analysis behaviors of `TryCast`.
-class TryCastSuite extends SparkFunSuite {
+class TryCastSuite extends CastWithAnsiOnSuite {
+
+  override def evalMode: EvalMode.Value = EvalMode.TRY
 
-  private def cast(v: Any, targetType: DataType, timeZoneId: Option[String] = None): TryCast = {
+  override def cast(v: Any, targetType: DataType, timeZoneId: Option[String] = None): Cast = {
     v match {
-      case lit: Expression => TryCast(lit, targetType, timeZoneId)
-      case _ => TryCast(Literal(v), targetType, timeZoneId)
+      case lit: Expression => Cast(lit, targetType, timeZoneId, EvalMode.TRY)
+      case _ => Cast(Literal(v), targetType, timeZoneId, EvalMode.TRY)
     }
   }
 
+  override def checkExceptionInExpression[T <: Throwable : ClassTag](
```

Comment:
this is a bit weird, I see code like this
```
    if (!isTryCast) {
      checkExceptionInExpression[ArithmeticException](ret, "overflow")
    } else {
      checkEvaluation(ret, Row(null))
    }
```

if we override `checkExceptionInExpression`, why do we still need `if-else`?

Reply:
That one was "Row(null)" with schema `struct<a:int>`. Different from the default answer `null`

Reply:
It's a bit weird to have two different testing approaches for try_cast, but given that it was already the case before, I don't have a strong opinion.